assets/questions/palindrome.js

```diff
@@ -0,0 +1,152 @@
+            chr = random.choice(chr_set)
+            if input_size == 1:
+                ret += chr
+                input_size -= 1
+            else:
+                ret = chr + ret + chr
+                input_size -= 2
+        return ret
+
+    @classmethod
+    def countNumberOfChar(cls, s):
+        ascii_codes = {i:0 for i in range(255)}
+        for chr in s:
+            ascii_codes[ord(chr)] += 1
+        for code in ascii_codes:
+            if ascii_codes[code] % 2 is not 0:
+                return False
+        return True
+
+    @classmethod
+    def forgetIgnoreSpace(cls, s):
+        for i in range(len(s)/2):
+            if s[i] is not s[-i-1]:
+                return False
+        return True
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'For this question, you will implement the isPalindrom function. ',
+        'It takes a sentense as input and return True if the input is a ',
+        'palindrome and False if it is not.'
+      ].join(''),
+      [
+        'A palindrom is a word, phrase, or sequence that reads the same ',
+        'backward as forward, e.g., \'madam\' or \'nurses run\'.'
+      ].join(''),
+      'Note, the input string would only consist of alphabets from a-z and space.'
+    ],
+    prerequisiteSkills: ['Strings', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'isPalindrome',
+    correctnessTests: [{
+      input: '',
+      allowedOutputs: [true]
+    }, {
+      input: 'nurses run',
+      allowedOutputs: [true]
+    }, {
+      input: 'a b   cba',
+      allowedOutputs: [true]
+    }, {
+      input: 'abcdcba',
+      allowedOutputs: [true]
+    }, {
+      input: 'abccba',
+      allowedOutputs: [true]
+    }, {
+      input: 'abab',
+      allowedOutputs: [false]
+    }, {
+      input: 'stressed desserts',
+      allowedOutputs: [true]
+    }, {
+      input: 'dioramas is samaroid',
+      allowedOutputs: [false]
+    }, {
+      input: 'dioramas is si samaroid',
+      allowedOutputs: [true]
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.countNumberOfChar',
+      messages: [
+        "Try running your code on 'abab' on paper. Did you expet that result?",
+        [
+          'Are you sure the sentence is actually palindrom? \'abba\' is palindrom, ',
+          'but \'abab\' is not.'
+        ].join(''),
+        [
+          "It looks like you're counting the number of character of the input string, ",
+          "and if you have the even number of each kind, returning true. That's not ",
+          "quite correct. See if you can figure out why."
+        ].join('')
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.forgetIgnoreSpace',
+      messages: [
+        "Try running your code on 'nurses run' on paper. Did you expect that result?",
+        [
+          'Try harder for cases with spaces. \'nurses run\' is palindrom, ',
```

Comment:
We should try to avoid overly critical language, since we're trying to encourage the student to do better, not critique them for writing a bug.

Suggested fix:

"What happens if your input string contains spaces?"